### PR TITLE
Add reason field

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward suggestion made in https://github.com/giantswarm/giantswarm/issues/5751
- Update reason when helm chart had been crashed.

### CODE REVIEW

File: service/controller/app/v1/resource/status/create.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		AppVersion: chartStatus.AppVersion,
 		Release: v1alpha1.AppStatusRelease{
 			LastDeployed: *chartStatus.Release.LastDeployed.DeepCopy(),
```
Comment (xh3b4sd) [inline service/controller/app/v1/resource/status/create.go]: FMI why does that need a deep copy?

Comment (tomahawk28) [reply service/controller/app/v1/resource/status/create.go]: I think it was @kopiczko who introduce this practice on chart-operator, can you tell me the reason?

Comment (kopiczko) [reply service/controller/app/v1/resource/status/create.go]: I don't remember me doing that and if I did I don't remember why :D

Comment (xh3b4sd) [reply service/controller/app/v1/resource/status/create.go]: When there is no reason we should drop it and keep it simple. [`LastDeployed`](https://godoc.org/github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1#AppStatusRelease) is only `DeepCopyTime`.

Comment (tomahawk28) [reply service/controller/app/v1/resource/status/create.go]: Do you mind if I open another PR for that, I would like to keep the subject like this

Comment (xh3b4sd) [reply service/controller/app/v1/resource/status/create.go]: Sure

**Pull request merged**